### PR TITLE
Add pdn model

### PR DESCRIPTION
**Pull request by wwxxzz**

### CODE REVIEW

File: easy_rec/python/model/pdn.py [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+    else:
+      self._bias_features = None
+
+    # user to item behavior info
+    u2i_seqs, _, _ = self._input_layer(
+        self._feature_dict, 'u2i_seq', is_combine=False)
+    self._seq_len = u2i_seqs[0][1]
+    self._u2i_seq = tf.concat([x[0] for x in u2i_seqs], axis=2)
+    # item side info
+    i_seqs, _, _ = self._input_layer(
```
Comment (poson) [inline easy_rec/python/model/pdn.py:44]: i_seqs, _, _ = self._input_layer(
        self._feature_dict, 'i_seq', is_combine=False)
    self._i_seq = tf.concat([x[0] for x in i_seqs], axis=2)

这部分3段都一样，可以封装成一个函数吧。

File: easy_rec/python/model/pdn.py [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+            axis=2,
+            separator=','),
+        axis=1,
+        separator=';')
+
+    # log(1 + exp(x)): map (-inf, inf) to (0, inf)
+    # logits = tf.nn.softplus(sim_out + trigger_out)
+    logits = tf.multiply(sim_out, trigger_out)
+
+    # add sequence mask
```
Comment (poson) [inline easy_rec/python/model/pdn.py:122]: 每个注释及其后面的大段代码应该封装为函数

File: easy_rec/python/model/pdn.py [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    trigger_merge_fea = trigger_seq_fea + user_fea[:, None, :]
+    trigger_dnn_layer = dnn.DNN(
+        self._model_config.trigger_dnn,
+        self._l2_reg,
+        'trigger_dnn',
+        self._is_training,
+        last_layer_no_activation=True,
+        last_layer_no_batch_norm=True)
+
+    # output: N x seq_len x d, d is usually set to 1
```
Comment (poson) [inline easy_rec/python/model/pdn.py:72]: 每个注释及其后面的大段代码应该封装为函数

File: docs/source/models/pdn.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+  - item: 物品侧相关特征，如 category, brand等
+  - bias: 导致选择性偏差的特征（位置信息等），如 position等
+- pdn: pdn模型相关的参数
+  - user_dnn：构造DNN编码user特征信息转换为向量
+  - item_dnn：构造DNN编码item特征信息转换为向量
+  - u2i_dnn：构造DNN编码user to trigger item的行为特征信息转换为向量
+  - i2i_dnn：构造DNN编码trigger item to target item特征转换为向量
+  - trigger_dnn：构造TrigNetDNN计算user对每个交互item的喜爱程度来捕获user的多峰兴趣
+  - sim_dnn：构造SimNetDNN基于item信息和共现信息计算trigger item与target item的相似度
+  - bias_dnn：基于会导致选择性偏差的特征（位置信息等）训练一个浅层DNN
```
Comment (poson) [inline docs/source/models/pdn.md:96]: bias_dnn 是否可以没有呢？

File: easy_rec/python/model/pdn.py (hunk 1 above)
Comment (wwxxzz) [reply easy_rec/python/model/pdn.py]: done. fixed.

File: easy_rec/python/model/pdn.py (hunk 2 above)
Comment (wwxxzz) [reply easy_rec/python/model/pdn.py]: done. fixed.

File: docs/source/models/pdn.md (hunk 4 above)
Comment (wwxxzz) [reply docs/source/models/pdn.md]: bias dnn可以没有的，修改为 "bias_dnn（可选）: …………"

**Review by tiankongdeguiji — APPROVED** [review]
LGTM
